File: src/dataset/cpp/atbash_cipher_Exercism_johnngugi/claude/ClaudeSonnet4_atbash_cipher_v4.cpp
#include <algorithm>
#include <string>
#include "atbash_cipher.h"

namespace atbash {

    char letters(char c) {
        if (c >= '0' && c <= '9') {
            return c;
        } else if (c >= 'a' && c <= 'z') {
            return 'z' - (c - 'a');
        } else if (c >= 'A' && c <= 'Z') {
            return 'z' - (c - 'A');
        }
        return c;
    }

    std::string cipher(std::string text) {
        std::string result;
        result.reserve(text.length());

        for (char c : text) {
            if ((c >= '0' && c <= '9') || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')) {
                result += letters(c);
            }
        }

        return result;
    }

    std::string encode(std::string text) {
        std::string store = cipher(std::move(text));
        std::string result;
        result.reserve(store.length() + store.length() / 5);

        for (size_t i = 0; i < store.length(); ++i) {
            result += store[i];

            if ((i + 1) % 5 == 0 && (i + 1) != store.length()) {
                result += ' ';
            }
        }

        return result;
    }

    std::string decode(std::string text) {
        return cipher(std::move(text));
    }
}

File: src/dataset/typescript/atbash-cipher_Exercism-typescript-shybyte/claude/ClaudeSonnet4_atbash-cipher_v4.ts
const aCode = 97
const zCode = 122

export default class AtbashCipher {
    encode = (s: string) => {
        let result = ''
        let count = 0
        
        for (let i = 0; i < s.length; i++) {
            const code = s.charCodeAt(i)
            
            if (code >= 48 && code <= 57) {
                result += s[i]
                count++
            } else if (code >= 65 && code <= 90) {
                result += String.fromCharCode(zCode - (code + 32 - aCode))
                count++
            } else if (code >= 97 && code <= 122) {
                result += String.fromCharCode(zCode - (code - aCode))
                count++
            }
            
            if (count === 5) {
                result += ' '
                count = 0
            }
        }
        
        return result.trim()
    }

    decode = (text: string): string => {
        let result = ''
        
        for (let i = 0; i < text.length; i++) {
            const code = text.charCodeAt(i)
            
            if (code >= 97 && code <= 122) {
                result += String.fromCharCode(zCode - code + aCode)
            } else if (code >= 48 && code <= 57) {
                result += text[i]
            }
        }
        
        return result
    }
}